src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -110,6 +111,9 @@ public Task<RestoreResult> ExecuteAsync()
                 }
             }
 
+            // Validate, for noop this will be replayed from the assets file.
+            _success &= await ValidateProjectAsync(_request.Project, _logger);
```

Comment:
Good question, I hadn't considered it wouldn't before. I looked it up and `&&` and `||` have lazy evaluation, but `&` and `|` do not.  So yes, it will always evaluate.

To skip it would need `&&=` which C# doesn't have. 
https://stackoverflow.com/questions/23107162/do-the-and-operators-for-bool-short-circuit